Sources/_CryptoExtras/RSA/RSA+BlindSigning.swift

```diff
@@ -0,0 +1,414 @@
+    /// randomized message preparation.
+    ///
+    /// - Seealso: [RFC 9474: RSABSSA Variants](https://www.rfc-editor.org/rfc/rfc9474.html#name-rsabssa-variants).
+    public static let RSABSSA_SHA384_PSSZERO_Deterministic = Self<SHA384>(padding: .PSSZERO, preparation: .identity)
+}
+
+extension _RSA.BlindSigning {
+    /// An input ready to be blinded, possibly prepended with random bytes.
+    ///
+    /// Users should not attempt to create values of this type; it is created and returned by the prepare operation.
+    public struct PreparedMessage {
+        var rawRepresentation: Data
+    }
+
+    /// The blinding inverse for a blinded messaeg, used to unblind a blind signature.
+    ///
+    /// Users should not attempt to create values of this type; it is created and returned by the blind operation.
+    public struct BlindInverse {
+        var rawRepresentation: Data
+    }
+}
+
+extension _RSA.BlindSigning.PrivateKey {
+    /// Generate a blind signature with the given key for a blinded message.
+    ///
+    /// - Parameter message: The blinded message to sign.
+    /// - Returns: A blind signature.
+    /// - Throws: If there is a failure producing the signature.
+    ///
+    /// - Seealso: [RFC 9474: BlindSign](https://www.rfc-editor.org/rfc/rfc9474.html#name-blindsign).
+    public func blindSignature<D: DataProtocol>(for message: D) throws -> _RSA.BlindSigning.BlindSignature {
+        try self.backing.blindSignature(for: message)
+    }
+}
+
+extension _RSA.BlindSigning.PublicKey {
+    /// Prepare a message to be signed using the blind signing protocol.
+    ///
+    /// - Parameter message: The message to be signed.
+    /// - Parameter parameters: Parameters used in the blind signing protocol.
+    /// - Returns: A preprared mesage, modified according to the parameters provided.
+    ///
+    /// - Seealso: [RFC 9474: Prepare](https://www.rfc-editor.org/rfc/rfc9474.html#name-prepare).
+    public func prepare<D: DataProtocol>(_ message: D) -> _RSA.BlindSigning.PreparedMessage {
+        switch self.parameters.preparation {
+        case .identity:
+            return _RSA.BlindSigning.PreparedMessage(rawRepresentation: Data(message))
+        case .randomized:
+            return _RSA.BlindSigning.PreparedMessage(rawRepresentation: Data(SystemRandomNumberGenerator.randomBytes(count: 32)) + message)
+        }
+    }
+
+    /// Blind a message to be signed by the server using the blind signing protocol.
+    ///
+    /// - Parameter message: The message to be signed.
+    /// - Returns: The blinded message, and its inverse for unblinding its blind signature.
+    ///
+    /// - Seealso: [RFC 9474: Blind](https://www.rfc-editor.org/rfc/rfc9474.html#name-blind).
+    public func blind(
+        _ message: _RSA.BlindSigning.PreparedMessage
+    ) throws -> (blindedMessage: Data, blindInverse: _RSA.BlindSigning.BlindInverse) {
```

Comment:
I'm generally opposed to using tuples as return values, and would prefer we used a nominal type here.

Reply:
I chose not to wrap these because we already have some wrapper types for the inverse and because the RFC function is returning two things.

I had a `BlindedMessage` type but this didn't work super well as an opaque type because the client needs to send just the message bytes (not the inverse, ofc) to the server. It also seemed that the result of running this protocol was to get to the point where you had a byte string and the blinding inverse.

I guess we could reinstate the `BlindedMessage` type but make it _not_ opaque:

```swift
struct BlindedMessage {
    public var blindedMessage: Data
    public var blindInverse: _RSA.BlindSigning.BlindInverse
}
```

What I don't love about this is (1) the nested repeated names and (2) I don't want API users to think they need to serialize and send this type (all of it) off to the server.

Do you have any suggestions for something clearer?

Reply:
My inclination is to call the return type `BlindingResult`, then have `blindedMessage` and `inverse`. That strikes me as the better spelling. The remaining concerns can be addressed with documentation.

Reply:
Done.